autoscheduler/frontend/src/tests/redux/CourseCards.test.ts

```diff
@@ -274,4 +286,108 @@ describe('Course Cards Redux', () => {
       });
     });
   });
+
+  describe('clearCourseCards', () => {
+    test('resets course cards to initial state', () => {
+      // arrange
+      const expected: CourseCardArray = {
+        numCardsCreated: 1,
+        0: {
+          course: '',
+          customizationLevel: CustomizationLevel.BASIC,
+          web: 'exclude',
+          honors: 'exclude',
+          sections: [],
+        },
+      };
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      // add another course card, should be removed by clearCourseCards()
+      store.dispatch(addCourseCard({}));
+
+      // act
+      store.dispatch(clearCourseCards());
+
+      // assert
+      expect(store.getState().courseCards).toEqual(expected);
+    });
+  });
+
+  describe('replaceCourseCards', () => {
+    test('replaces all course cards and keeps all properties not related to sections', async () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const expectedCourseCards: CourseCardArray = {
+        0: {
+          course: 'MATH 151',
+          customizationLevel: CustomizationLevel.BASIC,
+          web: 'no_preference',
+          honors: 'exclude',
+        },
+        numCardsCreated: 1,
+      };
+      const courseCards = [
+        {
+          course: 'MATH 151',
+          customizationLevel: CustomizationLevel.BASIC,
+          web: 'no_preference',
+          honors: 'exclude',
+        },
+      ];
+      fetchMock.mockImplementationOnce(testFetch);
+
+      // act
+      store.dispatch<any>(replaceCourseCards(courseCards, '202031'));
+      // wait for all actions to finish
+      await new Promise(setImmediate);
+
+      // assert
+      expect(store.getState().courseCards).toMatchObject(expectedCourseCards);
+    });
+
+    test('adds new sections for course cards', async () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const courseCards: SerializedCourseCardOptions[] = [
+        {
+          course: 'MATH 151',
+          customizationLevel: CustomizationLevel.BASIC,
+        },
+      ];
+      fetchMock.mockImplementationOnce(testFetch);
+
+      // act
+      store.dispatch<any>(replaceCourseCards(courseCards, '202031'));
+      // wait for all actions to finish
+      await new Promise(setImmediate);
+
+      // assert
+      // testFetch with a MATH course has 1 section, which should be added
```

Comment:
Do y'all think we need to add one more test to make sure that the `replaceCourseCards` action works if the `fetch`es return out of order? So, for example, if the `/api/sections` call returned card #1 before card #0, they stay in order? Intuitively, it feels like something that could go wrong if someone messed with it, but Ryan's implementation right now is solid enough to where it feels like common sense, so I'm not sure.

Reply:
I added one just so that we have more thorough tests, and just in case we change the implementation later we can still know that this works